src/storeReducers/cartReducer.js

```diff
@@ -41,18 +54,51 @@ export const getCart = (userLogin = false) => (dispatch, getStore) => {
           existingStore.cart.items.length &&
```

Comment:
These are busy functions, but I also don't really know how they can be streamlined. I don't think there is anything wrong with multiple dispatches since different aspects of the cart are controlled by different reducers.